### ISSUE TITLE
co: distinguish essential and optional consumables 

### ISSUE THREAD

**Issue by EvaJanouskova** [UCL/TLOmodel#1384]
Currently we don't distinguish essential and optional consumables for contraception methods, we assume all consumables are essential. This needs to be updated!

**Comment by EvaJanouskova** [comment]
@tbhallett, is this necessary to be done soon-ish, or could this wait?

**Comment by EvaJanouskova** [comment]
It might change the calibration but we should do it sometime but I think we could put it off until later.?

**Comment by EvaJanouskova** [comment]
If the calibration is based on default availability (Is it?), we made already a few changes which may slightly alternate the calibration within the PR #1298:

**master**
- one item for both type of pills (item code 1: `Ethinylestradiol 0.03mg + levonorgestrel 0.15mg_Each_FP000800_CMST, cycle //80% patients//; or Microlut (Levonorgestrel 0.03mg)_Each_FP004200_CMST, cycle //20% patients//`), hence availability based on item 1
- one item for both type of implants (item code 12: `Jadelle(implant)_Each_FP003700_CMST, 2 rods //50% patients//; or Implanon (Etonogestrel 68mg)_Each_FP004100_CMST, 1 rod //50% patients//`), hence availability based on item 12
- item 307: `Atropine sulphate 600 micrograms/ml, 1ml_Each_BB006600_CMST //1 unit 50% patients → 0.5 units per case//` - 0.5 units used, hence the availability always checked
- new items 2676: `Polyamide monofilament suture sterile 1, on 40mm 3/8 circle reverse cutting needle_12_GG005100_CMST` and 2677: C`atgut chromic suture sterile 0, 75cm, round bodied ½ circle 40mm needle_12_GG000600_CMST` used for female sterilisation

**#1298**
- two different items, one for each type of pills (item 0: `Levonorgestrel 0.0375 mg, cycle`, item 1: `Levonorgestrel 0.15 mg + Ethinyl estradiol 30 mcg (Microgynon), cycle`), hence the availability based on item 0 or item 1 according to which item is chosen by probability 20% : 80%
- two different items, one for each type of implant (item 12: `Jadelle (implant), box of 2_CMST`, item 13: `Implanon (Etonogestrel 68 mg`), hence the availability based on item 12 or item 13 according to which item is chosen by probability 50% : 50%
- item 307: `Atropine sulphate  600 micrograms/ml, 1ml_each_CMST` - used in 50% cases based on probability, hence the availability checked only in 50% cases of female sterilisation
- existing items 19: `Needle, suture, assorted sizes, round body` and 20:  `Suture, catgut, chromic, 0, 150 cm` used for female sterilisation

So maybe we should add the TODO to check the calibration in the #1298?

**Comment by EvaJanouskova** [comment]
> If the calibration is based on default availability (Is it?), we made already a few changes which may slightly alternate the calibration within the PR #1298:
> 
> **master**
> 
> * one item for both type of pills (item code 1: `Ethinylestradiol 0.03mg + levonorgestrel 0.15mg_Each_FP000800_CMST, cycle //80% patients//; or Microlut (Levonorgestrel 0.03mg)_Each_FP004200_CMST, cycle //20% patients//`), hence availability based on item 1
> * one item for both type of implants (item code 12: `Jadelle(implant)_Each_FP003700_CMST, 2 rods //50% patients//; or Implanon (Etonogestrel 68mg)_Each_FP004100_CMST, 1 rod //50% patients//`), hence availability based on item 12
> * item 307: `Atropine sulphate 600 micrograms/ml, 1ml_Each_BB006600_CMST //1 unit 50% patients → 0.5 units per case//` - 0.5 units used, hence the availability always checked
> * new items 2676: `Polyamide monofilament suture sterile 1, on 40mm 3/8 circle reverse cutting needle_12_GG005100_CMST` and 2677: C`atgut chromic suture sterile 0, 75cm, round bodied ½ circle 40mm needle_12_GG000600_CMST` used for female sterilisation
> 
> **#1298**
> 
> * two different items, one for each type of pills (item 0: `Levonorgestrel 0.0375 mg, cycle`, item 1: `Levonorgestrel 0.15 mg + Ethinyl estradiol 30 mcg (Microgynon), cycle`), hence the availability based on item 0 or item 1 according to which item is chosen by probability 20% : 80%
> * two different items, one for each type of implant (item 12: `Jadelle (implant), box of 2_CMST`, item 13: `Implanon (Etonogestrel 68 mg`), hence the availability based on item 12 or item 13 according to which item is chosen by probability 50% : 50%
> * item 307: `Atropine sulphate  600 micrograms/ml, 1ml_each_CMST` - used in 50% cases based on probability, hence the availability checked only in 50% cases of female sterilisation
> * existing items 19: `Needle, suture, assorted sizes, round body` and 20:  `Suture, catgut, chromic, 0, 150 cm` used for female sterilisation
> 
> So maybe we should add the TODO to check the calibration in the #1298?

_This is rolled back now in #1298._

**Comment by tbhallett** [comment]
@EvaJanouskova  can we close this?

**Comment by EvaJanouskova** [comment]
Don't you think the consumables for contraceptives should be distinguished as essential X optional in future along with calibration check/update? If not we can close this issue. However, if you agree that it should be done once, we may keep the issue open as a reminder this needs to be addressed.

**Comment by tbhallett** [comment]
Yes if you still think it's relevant (it will be indeed, if there are consumables currently labelled as being essential that are not actually essential), then please do keep it open and assign it to yourself.

**Comment by EvaJanouskova** [comment]
> Yes if you still think it's relevant (it will be indeed, if there are consumables currently labelled as being essential that are not actually essential), then please do keep it open and assign it to yourself.

Certainly, some items should be optional but were coded as essential.
